src/vaults/libraries/VaultPricingLib.sol

```diff
@@ -48,12 +55,16 @@ library VaultPricingLib {
     }
 
     /// @dev    Calculates share price and returns the value in price decimals
-    function calculatePrice(address vault, uint128 assets, uint128 shares) internal view returns (uint256) {
+    function calculatePrice(address shareToken, uint128 shares, address asset, uint256 tokenId, uint128 assets)
+        internal
+        view
+        returns (uint256)
```

Comment:
Question. Is there any reason why we're not using D18 here, given we force 18 decimals.

Reply:
I think the answer is that this library was used in v2 where D18 did not exist. IMO, it's out of scope for this PR to further refactor/optimize the `VaultPricingLib`. I will add it as an item to the cleanup list together with [your other comment](https://github.com/centrifuge/protocol-v3/pull/184/files#r2033956515).